lib/dor/was_crawl/warc_extractor_service.rb

```diff
@@ -37,6 +32,45 @@ def wacz_filepath
       def wacz_basename
         @wacz_basename ||= File.basename(wacz_filename, '.wacz')
       end
+
+      def wacz_data_package
+        Zip::File.open(wacz_filepath) do |wacz_file|
+          @wacz_data_package ||= JSON.parse(wacz_file.glob('datapackage.json').first.get_input_stream.read)
+        end
+      end
+
+      def data_package_profile
+        @data_package_profile ||= wacz_data_package['profile']
+      end
+
+      def data_package_resources
+        @data_package_resources ||= wacz_data_package['resources']
+      end
+
+      def extract_data_package
+        Zip::File.open(wacz_filepath) do |wacz_file|
+          wacz_file.glob('archive/*.warc.gz').each do |warc_entry|
+            filename = warc_entry.name.delete_prefix('archive/')
+            # Prefixing with WACZ filename to make unique.
+            warc_entry.extract(File.join(base_path, "#{wacz_basename}-#{filename}"))
+          end
+        end
+        File.delete(wacz_filepath)
+      end
+
+      def extract_multi_wacz_package
+        multi_wacz_filepath = wacz_filepath
+        Zip::File.open(multi_wacz_filepath) do |wacz_file|
+          data_package_resources.each do |resource|
```

Comment:
I wonder if we should ensure we only process `.wacz` files here? Perhaps other things might get bundled into multi-wacz files at some point?

Reply:
I did some testing and it doesn't look like cdxj-indexer throws an error with the screenshot and text WARC files. But perhaps we should ignore them so we aren't storing them in /web-archiving-stacks since they aren't used?

Reply:
The filtering of screenshots and text WARCs looks good to me!